### PR TITLE
Added a check of Untitled doc in _getNormalizedFilename and _getDenormalizedFilename

### PR DESCRIPTION
**Pull request by saurabh95**
Actually in the normalize path earlier there was no check of Untitled doc and it was appending the project dir to the Untitled path, so I added a check for Untitled doc before appending the project root.
Due to this JS codehints were not working for Untitled documents on changing the mode to JS.

All JavaScriptCodeHints tests are passing

### CODE REVIEW

**Review by swmitra — APPROVED** [review]

**Comment by ficristo** [comment]
Since this code will be refactored probably it is better to add a test.
Also you need to retarget the branch to release (if there aren't plans to merge again master to release of course).
Looking at the surrounding code of `tern-worker.js` usually parameters are passed to the functions instead of using a global variable like `isUntitledDoc` (see `handleUpdateFile` for example) but I don't think it is a big deal.
I added these two functions in https://github.com/adobe/brackets/pull/13147 in order to be able to remove `normalizeFilename` override. It's not pretty and if there are ways to improve the situation I would only be happy.

Overall this seems ok to me.

**Comment by saurabh95** [comment]
@ficristo I will definitely add test after the release activities. Also will try to refactor the code as well

**Comment by ficristo** [comment]
You can go ahead and merge for me.
Since I think you should retarget this PR to release branch I leave the merge to you.

**Comment by swmitra** [comment]
@saurabh95 I am fine with the changes to unblock us for the release now, but would like to see some tests being added in general for InMemory document use cases (not just JS code hints). Also note that, eventually we have to refactor and migrate this code to Tern on node as well so that we can merge that PR on master after brackets 1.9 release. We may need more changes to leverage on new features/APIs added in Tern.

**Comment by swmitra** [comment]
Merging.

**Review by petetnt — APPROVED** [review]
LGTM

**Comment by swmitra** [comment]
@petetnt We both were typing together I guess 😄

**Comment by swmitra** [comment]
@ficristo Let him do cherrypick after merging this to master, to keep master and release even for this module 👍

**Pull request merged**
